cEP-0022.md

```diff
@@ -0,0 +1,162 @@
+follow their own standards. We need to adapt to those standards with least
+user interaction and produce config files which generate absolutely no errors
+in their codebase. This will lead to easier adoption of coala across communities
+as it is going to result in an easy mergeable Pull Request regarding
+*adding `.coafile` and CI to enforce it* in the other orgs for example
+<https://github.com/processing/processing/pull/4781>.
+
+# Implementation
+
+This project will be implemented in two phases:
+
+- The Brute Force way
+- Further optimization on the Brute Force
+
+1. The first step will consist of preparing a dict of all the values that can
+   be attained by the bears' settings, may it be necessary or optional as we need
+   to create *green* config files that are as specific as possible. Failure to do
+   so will lead to dropping of most **optional** bear settings, thus will lead to
+   coala not detecting many inconsistencies, and the developers of the project will
+   have to do the work of learning about the bears and their settings and then
+   modifying the `.coafile` manually, while not able to guess the correct
+   **necessary** bear settings will lead to dropping of bears themselves.
+
+**The following steps mainly describe the brute force way**
+
+2. Certain thresholds may be defined for an org based on several factors
+   for the number of inconsistencies detected to be considered mistakes however
+   these shall be stored for later as the first and foremost goal of this project
+   is to generate completely *"green"* config files.
+
+3. coala will be ran over and over again for the entire project with all
+   instances of bears with all the combinations of settings running in
+   parallel, as soon as instances of a particular setting is done and we get
+   a value for the setting for which no errors were generated, it goes directly
+   into the `.coafile`, if no value for an optional setting matches, the setting will
+   have to be dropped,  while if no necessary setting is found out to be *green*,
+   the entire bear will have to be dropped, keeping at the back (of our heads) that,
+   certain settings producing number of errors within the threshold as described in
+   step 2 are being saved for later.
+
+4. Once we have collected this data and using the metadata already being
+   collected from `coala-quickstart` we will generate our first and foremost complete
+   *green* config file.
+
+5. The additional data generated for erraneous settings within a theshold can
+   be reported later to the project administrators in any form and wait for their
+   decision whether these are actual codestyles or mistakes. In case of mistakes,
+   `coala-quickstart` should have the ability to add these rules automatically
+   to the config file. In case these are actual codestyles, `coala-quickstart`
+   should store this data for use in the next phase.
+
+# Improving upon the Brute Force
```

Comment:
Another idea:

Instead of exhaustive search to find out the right value, we can explore the possibility of using more sophisticated search algorithms. https://en.wikiversity.org/wiki/Search_techniques